app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/providers/Retrofit.kt

```diff
@@ -0,0 +1,88 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network.providers
+
+import android.util.Log
+import com.google.gson.Gson
+import com.naxtlevelofandroiddevelopment.marvelgallery.BuildConfig
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.math.BigInteger
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.concurrent.TimeUnit
+
+object RetrofitApi : Provider<Retrofit>({
```

Comment:
this class does too much. I would delegate responisibilities logically for example a separate class for handling security stuff